### ISSUE TITLE
Best way to avoid `Assertion error` 

### ISSUE THREAD

**Issue by tduretz** [JuliaSIMD/LoopVectorization.jl#345]
Dear LV developpers,

I am trying to run a simulation in which I call a function which takes (way too) many input arguments inside a loop.
If I use @turbo, I get an Assertion Error: N <= 16. 
This basically tells me that my function has too many arguments for LV to digest it, right? 

One work around would then be to group several of the scalar inputs into a named tuple, passed it as a single argument instead of N arguments. This would allow to keep the total number of arguments <=16. However, using a name tuple returns

```
LoadError: Expression not recognized.
```

A MWE would be this:
```julia

function main()

    N = 100;
    a1 = ones(N) # for the MWE I need at least 1 vectoir argument for LV otherwise check_args() fails
    a2,a3,a4,a5,a6,a7,a8,a9,a10,a11,a12,a13,a14,a15,a16,a17 = ones(16)

    # Works
    for i=1:N
        A = func(a1[i],a2,a3,a4,a5,a6,a7,a8,a9,a10,a11,a12,a13,a14,a15,a16,a17)
    end

    # Assertion error
    @turbo for i=1:N
        A = func(a1[i],a2,a3,a4,a5,a6,a7,a8,a9,a10,a11,a12,a13,a14,a15,a16,a17)
    end

    # Works since N<=16
    @turbo for i=1:N
        A = func16(a1[i],a2,a3,a4,a5,a6,a7,a8,a9,a10,a11,a12,a13,a14,a15,a16)
    end

    # try to use tuple to hide bad guys --> Expression not recognized.
    @turbo for i=1:N
        a = a2,a3,a4,a5,a6,a7,a8,a9,a10,a11,a12,a13,a14,a15,a16,a17
        A = func_tuple(a1[i],a)
    end
end

main()
```

I have used: 
```
  [bdcacae8] LoopVectorization v0.12.84
```

Any thought of some potential workarounds?

Cheers and thanks!

**Comment by chriselrod** [comment]
I guess LV could also be modified to remove the limit, with a bit more work.

**Comment by tduretz** [comment]
Thanks for the feedback. I could indeed try to inline the function. 
In my case input params a1 -> a17 are read from 2D tables.
Maybe I could simply change the data structure to make an array of, say 17, tables.
Then it would restrict the number of arguments (and make the code more readable).
I'll try to either inline or change data structure. I was just not aware about the limit of arguments when I started coding.
I think we can close this issue since I realise the issue is rather related to the way I structured the data/code :)
Thanks again!
